marker_uilints/tests/ui/expr/sugar/for_loop.stderr

```diff
@@ -0,0 +1,82 @@
+warning: expr
+  --> $DIR/for_loop.rs:9:5
+   |
+9  | /     for i in range {
+10 | |         total += i;
+11 | |     }
+   | |_____^
+   |
+   = note: SpanSource: File(
+               FileInfo {
+                   file: "$DIR/for_loop.rs",
+                   span_src: SpanSrcId(..),
+               },
+           )
+   = note: Snippet: "for i in range {/n        total += i;/n    }"
+note: the lint level is defined here
+  --> $DIR/for_loop.rs:8:12
+   |
+8  |     #[warn(marker::print_every_expr)]
+   |            ^^^^^^^^^^^^^^^^^^^^^^^^
+
+warning: expr
+ --> $DIR/for_loop.rs:9:14
+  |
+9 |     for i in range {
+  |              ^^^^^
+  |
+  = note: SpanSource: File(
+              FileInfo {
+                  file: "$DIR/for_loop.rs",
+                  span_src: SpanSrcId(..),
+              },
+          )
+  = note: Snippet: "range"
+
+warning: expr
+  --> $DIR/for_loop.rs:9:20
+   |
+9  |       for i in range {
+   |  ____________________^
+10 | |         total += i;
+11 | |     }
+   | |_____^
```

Comment:
I didn't think that the block of the `for` is a separate expression. Is this right?

Reply:
Yeah, that's how it's defined in the reference: [Iterator loops](https://doc.rust-lang.org/reference/expressions/loop-expr.html#iterator-loops)

It also makes the conversion and util code easier to reuse structs